### PR TITLE
[Pytorch] pytorch only timesteps

### PR DESCRIPTION
**Pull request by kashif**
timesteps are all torch tensors now
made the `set_timesteps` API consistent

as opposed to discussion in #717

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py [hunk 1]
```diff
@@ -344,8 +344,11 @@ def __call__(
         t_start = max(num_inference_steps - init_timestep + offset, 0)
 
         # Some schedulers like PNDM have timesteps as arrays
-        # It's more optimzed to move all timesteps to correct device beforehand
-        timesteps_tensor = torch.tensor(self.scheduler.timesteps[t_start:], device=self.device)
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:348]: I think:

```
 timesteps_tensor = torch.tensor(self.scheduler.timesteps[t_start:], device=self.device)
```

was the better API - i.e. I think we should try to get rid of the `if else` statement by forcing everything to be in PyTorch

Comment (patrickvonplaten) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:348]: Could we try to not use the if-else statement here?

**Comment by patrickvonplaten** [comment]
Cool is this maybe related to: https://github.com/huggingface/diffusers/issues/728

**Comment by kashif** [comment]
@patrickvonplaten thank you!

**Comment by patrickvonplaten** [comment]
All slow tests are passing -> merging!

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**